media/js/teach/smarton.js

```diff
@@ -0,0 +1,300 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+;(function($) {
+    'use strict';
+
+    var $window = $(window);
+    var $document = $(document);
+    var $body = $('body');
+    var $header = $('.page-header');
+    var $firstSection = $('#ask');
+    var $firstStep = $('#nav-step-ask');
+    if ($header.length > 0) {
+        var headTop = $header.offset();
+        var headHeight = $header.height();
+    }
+
+    if (window.isFirefox()) {
+        $body.addClass('is-firefox');
+    } else {
+        $body.addClass('not-firefox');
+    }
+
+    // Sticky navigation
+    var fixed = false;
+    var didScroll = false;
+
+    $window.scroll(function() {
+        didScroll = true;
+    });
+
+    if ($header.length > 0) {
+        $(document).ready(function() {
+            var scrollTop = $window.scrollTop();
+            if (scrollTop >= headTop.top) {
+                didScroll = true;
+            }
+        });
+    }
+
+    function adjustScrollbar() {
+        if (didScroll) {
+            didScroll = false;
+            var scrollTop = $window.scrollTop();
+            if ($header.length > 0) {
+                if( scrollTop >= headTop.top ) {
+                    if(!fixed) {
+                        fixed = true;
+                        $header.addClass('fixed');
+                        $firstStep.addClass('current');
+                        // recalculate height after sticking
+                        headHeight = $header.height();
+                        // then pad the first section so it doesn't jump as far
+                        $firstSection.css({ paddingTop: headHeight });
+                    }
+                } else {
+                    if(fixed) {
+                        fixed = false;
+                        $header.removeClass('fixed');
+                        $firstStep.removeClass('current');
+                        $firstSection.removeAttr('style');
+                        $header.find('.nav-steps li').removeClass();
+                    }
+                }
+            }
+        }
+    }
+
+    // Check for an adjusted scrollbar every 100ms.
+    setInterval(adjustScrollbar, 100);
+
+    // Scroll smoothly to the linked section
+    $document.on('click', '.nav-steps a[href^="#"]', function(e) {
+        e.preventDefault();
+        // Extract the target element's ID from the link's href.
+        var elem = $(this).attr('href').replace( /.*?(#.*)/g, '$1' );
+        Mozilla.smoothScroll({
+            top: $(elem).offset().top - headHeight
+        }, function() {
```

Comment:
Hmm, smoothScroll doesn't accept a callback function like this, so you may have to rework this logic a little?